src/pages/home/base/mobile-core/platform-services/tabs/network-service.md

```diff
@@ -137,22 +174,42 @@ class MyCustomNetworkService implements Networking {
 
     private final ExecutorService executorService = //
 
+    /**
+     * Perfmoring an asynchronous network request.
+     *
+     * @param request {@link NetworkRequest} used for network connection
+     * @param callback {@link NetworkCallback} that will receive the {@link HttpConnecting} instance asynchronously.
+     *
+     */
     @Override
     public void connectAsync(final NetworkRequest request, final NetworkCallback callback) {
         // Use an executor service for initiating the network request and dispatching the response.
         executorService.submit(
-                () -> {
+            () -> {
+                try {
+                // 1. If the network is down, for example the device is under airplane mode, callback should be invoked immediately with a null connection.
```

Comment:
// 1. If the network is down, for example the device is under airplane mode, callback should be invoked immediately with a null connection. When the null connection is passed to the callback, the SDK will retry the same request at a later time.

                    // callback.call(null);